regression/contracts/quantifiers-loop-02/test.desc

```diff
@@ -1,6 +1,6 @@
-CORE smt-backend
+CORE smt-backend broken-cprover-smt-backend
 main.c
---apply-loop-contracts _ --z3
+--apply-loop-contracts _
```

Comment:
I understand the tag to mark this is broken for the `cprover` backend, but removing the `--z3` tag hides that this uses functionality explicit to `z3`. I know the default at the moment is `z3`, but leaving this explicit has clarity that I'm not sure we want to remove.

Reply:
Unless I have radically misunderstood, this should work with any SMT solver that supports quantifiers.  I will re-add --z3 but this means that the `smt-backend` tests will fail if another solver is used and `z3` is not present.

Reply:
Not for this specific example, but a lot of the quantified output is a problem for other solvers. Typically `cvc3` and `cvc4` will error out due to quantifiers while specifying quantifier free logic, similar behaviour for `mathsat` and `boolector`. I think only `z3` can handle it out of the box from `cbmc`, while modifying the `set-logic` command allows `cvc4` to handle it (but can produce unknowns... which we were avoiding until it became clear this happens for `z3` and is now handled gracefully). Also in the current backend the output for `z3` is different to most other solvers, so turning of the `--z3` flag may reveal different `SMT` code.

Reply:
> due to quantifiers while specifying quantifier free logic

This is the intended and correct behaviour of the solver.

> but can produce unknowns... which we were avoiding until it became clear this happens for z3 and is now handled gracefully

Again, if the logic you have requested / fragment of logic you have used is undecideable, this is intended behaviour.

Reply:
Apologies if my comment was unclear. I was not implying the behaviour was incorrect, only commenting on what we see now (and why I think being explicit on `z3` is helpful). No requested change or displeasure with the PR.

Reply:
Sorry about the late response on this, and for reopening this conversation.

I guess, until #6248 is merged, we could keep the `_ --z3` arguments, but it doesn't really matter for the CI because we don't run CVC4 anyway and so this test wouldn't really break anything (without the `_ --z3` part).
It would be nice to aim for solver-agnostic SMT encoding for as many cases as possible, and I think in this case we do get a solver-agnostic encoding that should work with both Z3 and CVC4 (modulo the `set-logic` issue). To me, `_ --z3` indicates that this test triggers some Z3-specific encoding that wouldn't work on other SMT solvers in general.

Moreover, if that's the case, we should use the `broken-<solver-X>-backend` tags to explicitly document that at least for Z3, CVC4 and CPROVER SMT2 solvers.

What do you think?

Reply:
TBH for the purposes of this test all I want is to be able to run the regression tests on a machine that doesn't have Z3 installed.  For me anything else is a discussion for another ticket / issue.

Reply:
Approving this and unblocking you. We can revisit this issue once #6248 is merged may be.